dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -552,7 +557,25 @@ CFileDefExt::InitFileAttr(HWND hwndDlg)
             FileSize.u.LowPart = FileInfo.nFileSizeLow;
             FileSize.u.HighPart = FileInfo.nFileSizeHigh;
             if (SH_FormatFileSizeWithBytes(&FileSize, wszBuf, _countof(wszBuf)))
-                SetDlgItemTextW(hwndDlg, 14011, wszBuf);
+            {    
+                SetDlgItemTextW(hwndDlg, 14011, wszBuf);                
+                // Calculate size on disc
+                // if disk usage is not aligned on a complete disk geometry block (sector/cluster), align on the next block size, depending on disk geometry data
+                // This could be improved as per : https://docs.microsoft.com/en-us/previous-versions/technet-magazine/hh148159(v=msdn.10)
+                if (GetVolumePathName(m_wszPath, szVolumePathName, _countof(szVolumePathName)))
+                {
+                    if (GetDiskFreeSpace(szVolumePathName, &ulSectorsPerCluster, &ulBytesPerSector, NULL, NULL))
+                    {
+                        ulBlockSize = ulBytesPerSector * ulSectorsPerCluster;
+                        if (FileSize.QuadPart % ulBlockSize)
+                        {
+                            FileSize.QuadPart = ((FileSize.QuadPart / (ulBlockSize)) + 1) * ulBlockSize;
```

Comment:
Your calculation is wrong. When file size was ulBlockSize, then the results will be 2 * ulBlockSize.

Reply:
Good point. I'll check and fix.

Reply:
Will be refactored using NtQueryInformationFile with FileStandardInformation

Reply:
Please remove this change request. Now obsolete.